### PR TITLE
Update Desktop Entry and AppStream metadata files

### PR DESCRIPTION
**Pull request by Vulcalien**
I updated the `.desktop` and `.metainfo.xml` files, fixing a typo (`pplication/x-agb-rom`), adding the new releases to the changelog etc

I also changed the project license from `GPL-3.0` to `GPL-3.0+` (tell me if I'm wrong, I think NBA is GPL 3 *or any later version*).

### CODE REVIEW

File: src/platform/qt/rc/io.nanoboyadvance.NanoBoyAdvance.metainfo.xml [hunk 1]
```diff
@@ -29,14 +29,64 @@ It aims to be as accurate as possible, while also offering enhancements such as
       <li>Passes all AGS aging cartridge tests (NBA was the first public emulator to achieve this)</li>
       <li>Passes most tests in the mGBA test suite</li>
       <li>Passes ARMWrestler, gba-suite and FuzzARM CPU tests</li>
+      <li>Very high compatibility, including games that require emulation of peculiar hardware edge-cases</li>
     </ul>
   </description>
   <screenshots>
     <screenshot type="default">
-      <image type="source">https://github.com/nba-emu/NanoBoyAdvance/raw/v1.7/docs/screenshot.png</image>
+      <image type="source">https://raw.githubusercontent.com/nba-emu/NanoBoyAdvance/master/docs/screenshot.png</image>
```
[the remark this answers is not in the document]
Comment (Vulcalien) [reply src/platform/qt/rc/io.nanoboyadvance.NanoBoyAdvance.metainfo.xml:37]: Yep, I was told that in the flatpak submission too. After the work there is done, the changes will be put here too.

**Comment by fleroviux** [comment]
> I also changed the project license from GPL-3.0 to GPL-3.0+ (tell me if I'm wrong, I think NBA is GPL 3 or any later version).

That's correct yes

File: src/platform/qt/rc/io.nanoboyadvance.NanoBoyAdvance.desktop [hunk 2]
```diff
@@ -1,12 +1,12 @@
-# Created with jdDesktopEntryEdit 1.0
 [Desktop Entry]
 Type=Application
 Name=NanoBoyAdvance
+GenericName=Emulator
```
Comment (zefr0x) [inline src/platform/qt/rc/io.nanoboyadvance.NanoBoyAdvance.desktop:4]: Isn't that too generic?

For example `Web Browser` and `File Browser` are Generic Names, so this should be `SomeThing Emulator`?

Comment (Vulcalien) [reply src/platform/qt/rc/io.nanoboyadvance.NanoBoyAdvance.desktop]: Changed to `Game Boy Advance Emulator`, thanks!

**Comment by Vulcalien** [comment]
I updated the files to the latest version in the flathub submission. I don't think they'll change anymore soon.

**Comment by fleroviux** [comment]
@Vulcalien Thanks for your efforts. Should I go ahead and merge this already?

**Comment by Vulcalien** [comment]
> Should I go ahead and merge this already?

Yep, I'm done with these changes
